pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -146,6 +146,10 @@ public void close() throws Exception {
 
     private boolean isCompatible(SchemaAndMetadata existingSchema, SchemaData newSchema,
                                  SchemaCompatibilityStrategy strategy) {
+        if (SchemaType.isPrimitiveSchemaType(existingSchema.schema.getType())) {
+            // for primitive data types, only schema type check is needed
+            return existingSchema.schema.getType() == newSchema.getType();
```

Comment:
@srkukarni even from primitive types you still should check this condition:
```
compatibilityChecks.getOrDefault(newSchema.getType(), SchemaCompatibilityCheck.DEFAULT)
            .isCompatible(existingSchema.schema, newSchema, strategy);	            .isCompatible(existingSchema.schema, newSchema, strategy)
```

Reply:
@jerrypeng fixed